gcloud-java-dns/src/main/java/com/google/gcloud/dns/DnsRecord.java

```diff
@@ -0,0 +1,280 @@
+     * Builds the DNS record.
+     */
+    public DnsRecord build() {
+      return new DnsRecord(this);
+    }
+
+    /**
+     * Sets references to the managed zone that this DNS record belongs to.
+     *
+     * todo(mderka): consider if this method is needed; may not be possible when listing records
+     */
+    Builder managedZone(ManagedZoneInfo parent) {
+      checkNotNull(parent);
+      this.zoneId = parent.id();
+      this.zoneName = parent.name();
+      return this;
+    }
+
+    /**
+     * Sets name reference to the managed zone that this DNS record belongs to.
+     */
+    Builder managedZone(String managedZoneName) {
+      this.zoneName = checkNotNull(managedZoneName);
+      return this;
+    }
+  }
+
+  /**
+   * Creates a builder pre-populated with the attribute values of this instance.
+   */
+  public Builder toBuilder() {
+    return new Builder(this);
+  }
+
+  /**
+   * Creates an empty builder
+   */
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  /**
+   * Get the mandatory user assigned name of this DNS record.
+   */
+  public String name() {
+    return name;
+  }
+
+  /**
+   * Returns a list of DNS record stored in this record set.
+   */
+  public List<String> rrdatas() {
+    return rrdatas;
+  }
+
+  /**
+   * Returns the number of seconds that this DnsResource can be cached by resolvers. This number is
+   * provided by the user.
+   */
+  public Integer ttl() {
+    return ttl;
+  }
+
+  /**
+   * Returns the type of this DNS record.
+   */
+  public DnsRecordType type() {
+    return type;
+  }
+
+  /**
+   * Returns name of the managed zone that this record belongs to. The name of the managed zone is
+   * provided by the user when the managed zone is created. It is unique within a project. If this
+   * DNS record is not associated with a managed zone, this returns null.
+   */
+  public String zoneName() {
+    return zoneName;
+  }
+
+  /**
+   * Returns id of the managed zone that this record belongs to.
+   *
+   * <p>The id of the managed zone is determined by the server when the managed zone is created. It
+   * is a read only value. If this DNS record is not associated with a managed zone, or if the id of
+   * the managed zone was not loaded from the cloud service, this returns null.
+   */
+  public Long zoneId() {
+    return zoneId;
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(name, rrdatas, ttl, type, zoneName, zoneId);
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    return (obj instanceof DnsRecord) && Objects.equals(this.toPb(), ((DnsRecord) obj).toPb())
+            && this.zoneId().equals(((DnsRecord) obj).zoneId())
+            && this.zoneName().equals(((DnsRecord) obj).zoneName());
+
+  }
+
+  com.google.api.services.dns.model.ResourceRecordSet toPb() {
+    com.google.api.services.dns.model.ResourceRecordSet pb =
+            new com.google.api.services.dns.model.ResourceRecordSet();
+    pb.setName(this.name());
```

Comment:
Is name mandatory?  Is the name going to be used for matching when used in a delete change request?